apps/meteor/app/apps/server/bridges/rooms.ts

```diff
@@ -245,6 +245,64 @@ export class AppRoomBridge extends RoomBridge {
 		return users.map((user: ICoreUser) => userConverter.convertToApp(user));
 	}
 
+	protected async getUnreadByUser(roomId: string, userId: string, options: GetMessagesOptions, appId: string): Promise<Array<IMessageRaw>> {
+		this.orch.debugLog(`The App ${appId} is getting the unread messages for the user: "${userId}" in the room: "${roomId}"`);
+
+		const messageConverter = this.orch.getConverters()?.get('messages');
+		if (!messageConverter) {
+			throw new Error('Message converter not found');
+		}
+
+		const [room, subscription] = await Promise.all([
+			Rooms.findOneById(roomId, { projection: { _id: 1 } }),
+			Subscriptions.findOneByRoomIdAndUserId(roomId, userId, { projection: { ls: 1 } }),
+		]);
```

Comment:
This is an interesting approach, I'm only concerned about "wasted" cases, i.e. if the room does not exist, a subscription to it would not exist as well - which means the query for a subscription is useless.

@MarcosSpessatto do you have any thoughts on this?

Reply:
Some stats for better tradeoff decision (for a count of 136 unread messages).

Concurrent approach:
```
Execution Times (excluding first run): 1.965ms, 5.788ms, 2.717ms, 1.631ms, 2.808ms
Average = (1.965 + 5.788 + 2.717 + 1.631 + 2.808) / 5 = 2.9818ms
```

Sequential Approach:
```
Execution Times (excluding first run): 9.735ms, 7.949ms, 4.642ms, 5.005ms, 7.303ms, 3.948ms, 4.997ms
Average = (9.735 + 7.949 + 4.642 + 5.005 + 7.303 + 3.948 + 4.997) / 7 = 6.2256ms
```

Concurrent approach is roughly twice as fast as the sequential approach.

Reply:
Yes, it will be faster, I'm more concerned with resource consumption and high DB loads - cpu cycles are "wasted" waiting on the promise and it needlessly contributes to he workload of the database (in those specific cases). However, the queries you're doing use indexes and transfer little data.

I might be overthinking this, I'll just get a third opinion here just in case.

Reply:
Well, interesting, it makes sense; I will use the sequential approach here.